02_eval/eval_pipeline.py

```diff
@@ -45,14 +76,37 @@ async def run_evaluation_pipeline(dataset, args):
 
 
 class EvaluationPipeline(GenerationPipeline):
-    def __init__(self):
+    """ 
+    Extension of a GenerationPipeline to generate, modify, then 
+    score the returned results from Lamini.generate.
+
+    Parameters
+    ----------
+    None
+
+    """
+
+    def __init__(self) -> None:
         super().__init__()
 
         self.model_gen_stage = LaminiModelStage()
         self.modify_stage = ModifyStage()
         self.score_stage = ScoreStage()
 
-    def forward(self, x):
+    def forward(self, x: PromptObject) -> PromptObject:
```

Comment:
forward() should be:
1. accepting Generator[PromptObject] or AsyncGenerator[PromptObject]
2. returning AsyncGenerator[PromptObject]

forward() today is a process of calling the GenerationNode subclasses' __call__ method (__call__ eventually invoke generate())

Reply:
Updated!